src/placement/jump_map.c

```diff
@@ -365,28 +367,35 @@ get_target(struct pool_domain *curr_dom, struct pool_target **target,
 			end_tgt = start_tgt + (num_doms - 1);
 
 			range_set = isset_range(tgts_used, start_tgt, end_tgt);
-			if (range_set)
-				clrbit_range(tgts_used, start_tgt, end_tgt);
-
-			do {
-				/*
-				 * Must crc key because jump consistent hash
-				 * requires an even distribution or it will
-				 * not work
-				 */
-				obj_key = crc(obj_key, fail_num++);
-
-				/* Get target for shard */
-				selected_dom = jump_consistent_hash(obj_key,
-								    num_doms);
+			if (range_set) {
+				/* Used up all targets in current domain,
+				 * let's try parent domain
+ 				 */
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs